api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,15 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
+				&& ((this.getEndDate() != null && o.getEndDate() != null
+				&& this.getEndDate().equals(o.getEndDate()))
+				|| (this.getEndDate() == null && o.getEndDate() == null))) {
+			return 0;
+		} else {
+			return -1;
+		}
```

Comment:
Doesn't matter really how you implement it, remember that the memberships are added to a TreeSet, so it would be nice to implement compareTo in a standard way where you return 1, 0, -1 and also have unit tests to back it up